### PR TITLE
examples: networking support for default example

### PR DESCRIPTION
**Pull request by OlegHahm**
~~Depends on #3800~~

### CODE REVIEW

File: examples/default/main.c [hunk 1]
```diff
@@ -47,6 +52,14 @@ int main(void)
     rtc_init();
 #endif
 
+#ifdef MODULE_GNRC
```
Comment (authmillenon) [inline examples/default/main.c]: Since `pktdump` is the top of the dependency tree here, I would take `MODULE_GNRC_PKTDUMP` here

**Comment by kaspar030** [comment]
I'd not include gnrc into default. examples/default should run on *all* platforms. gnrc does not fit everywhere.

**Comment by OlegHahm** [comment]
@kaspar030, we certainly need (at least) link layer networking in one example (and I would argue to put it into default as it was in former days). If you show me an easy way to add it without gnrc, I'm fine, too.

File: examples/default/main.c (hunk 1 above)
Comment (OlegHahm) [inline examples/default/main.c]: Hm, true, but in fact that's not what I really want here. I want to check if the current platform has gnrc support. Looking at the dependencies we could check for `MODULE_GNRC_NOMAC` here.

**Comment by kaspar030** [comment]
@OlegHahm Well, I agree. Does this PR's "default" compile for all platforms? I just realized that you're only using the link layer, so it might fit.

**Comment by OlegHahm** [comment]
@kaspar030, I checked with `make buildtest` and didn't see any failure, hence, I guess it fits on the same platforms as it did without this PR.

File: examples/default/main.c (hunk 1 above)
Comment (haukepetersen) [inline examples/default/main.c]: Thinking about this -> I think gnrc support is not what we actually strive for here, is it? I think that actual function we want to include into the default example is simple link-layer access to any available network device (in the form of ifconfig and txtsend). Right now we have those implemented through the gnrc stack, but with netdev2 I see on the mid-term actually no need to include any gnrc specifics, rather I say we probably can make `txtsend` and `ifconfig` stack agnostic?!

Comment (OlegHahm) [inline examples/default/main.c]: Probably yes, but for now (before the release) I would just use the gnrc lower layer functionalities.

**Comment by OlegHahm** [comment]
rebased to master

**Comment by OlegHahm** [comment]
updated

**Comment by OlegHahm** [comment]
@kaspar030, @haukepetersen, @authmillenon, an ACK anyone?

**Comment by authmillenon** [comment]
Doesn't the Readme needs update too?

**Comment by OlegHahm** [comment]
True - will do. Thanks for the pointer.

**Comment by OlegHahm** [comment]
Rebased and updated README

**Comment by authmillenon** [comment]
Works as written. Please squash. ACK.

**Comment by OlegHahm** [comment]
squashed

**Pull request merged**